src/app/core/models/repo.model.ts

```diff
@@ -16,13 +16,36 @@ export class Repo {
   public static of(repoUrl: string) {
     const repoUrlSplit = repoUrl.split('/');
     if (repoUrlSplit.length !== 2) {
-      return undefined; // throw error?
+      throw new Error('Invalid repo name. Please provide repo name in the format Org/Repo.');
     }
     return new Repo(repoUrlSplit[0], repoUrlSplit[1]);
   }
 
+  public static ofEmptyRepo() {
+    return EMPTY_REPO;
+  }
+
+  public static isEmptyRepo(repo: unknown) {
```

Comment:
I think ```isInvalidRepoName``` is more suitable here since it returns false when it is not a repo as well. Also makes more sense when it is called in ```src/app/shared/layout/header.component.ts```